dev/ApplicationData/ApplicationData.idl

```diff
@@ -0,0 +1,190 @@
+        String SharedLocalPath { get; };
+
+        /// Return the path for the temporary data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.temporaryfolder
+        String TemporaryPath { get; };
+
+        /// Return a StorageFolder for the local cache data store not included in backup and restore operations.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localcachefolder
+        Windows.Storage.StorageFolder LocalCacheFolder { get; };
+
+        /// Return a StorageFolder for the local data store. This location is backed up to the cloud.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localfolder
+        Windows.Storage.StorageFolder LocalFolder { get; };
+
+        /// Return a StorageFolder for the machine data store.
+        /// @note This location is per-machine (not per-user like other *Folder properties).
+        Windows.Storage.StorageFolder MachineFolder { get; };
+
+        /// Return a StorageFolder for the roaming data store.
+        /// @warning Roaming data and settings are no longer supported after Windows 10 version 1901 (aka 19H1 aka 10.0.18363.0).
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingfolder
+        Windows.Storage.StorageFolder RoamingFolder { get; };
+
+        /// Return a StorageFolder for the shared data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.sharedlocalfolder
+        Windows.Storage.StorageFolder SharedLocalFolder { get; };
+
+        /// Return a StorageFolder for the temporary data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.temporaryfolder
+        Windows.Storage.StorageFolder TemporaryFolder { get; };
+
+        /// Return the settings container in the local data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localsettings
+        ApplicationDataContainer LocalSettings { get; };
+
+        /// Return the settings container in the roaming data store.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingsettings
+        ApplicationDataContainer RoamingSettings { get; };
+
+        /// Remove all data from the local, local cache, roaming, and temporary data stores.
+        /// @see LocalCachePath
+        /// @see LocalCacheFolder
+        /// @see LocalPath
+        /// @see LocalFolder
+        /// @see RoamingPath
+        /// @see RoamingFolder
+        /// @see TemporaryPath
+        /// @see TemporaryFolder
+        [method_name("ClearAsync")]
+        Windows.Foundation.IAsyncAction ClearAllAsync();
+
+        /// Remove all data from the specified data store.
+        Windows.Foundation.IAsyncAction ClearAsync(ApplicationDataLocality locality);
```

Comment:
Is this `method_name` attribute needed?

I can't remember how does this work out in terms of ABI. We have two methods here that are going to want to be `ClearAsync`--is that expected? (Not all languages support overloading.)

Reply:
Yep. ByDesign.

Overload by arity so overloading works but ABI has different names (since WinRT is COM at the ABI level and COM doesn't support method overloading). How that's expressed by language projections depends on the projection, e.g. C++/WinRT supports method overloading so callers see

```
ClearAsync()
ClearAsync(Locality)
```

as they'd expect. For languages that don't support method overloading I'd expect they'd expose the ABI method names but that's really up to the projection tech.

P.S. I never remember if the method is the ABI and method_name() is the overloaded name or vice verse but MIDL takes this fine and errors the other way so it's hard to get it wrong :P

Reply:
Right, but in terms of ABI, I think:
* ClearAllAsync requests emission as `ClearAsync`
* ClearAsync default requests emission as `ClearAsync`

So it seems like this creates a conflict that might be handled differently per projection. (I haven't tried it yet.)